src/main/resources/bin/validate-parallel.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+#
+# Script that assists with splitting up PDS-4 products into multiple
+# groups, so that they can be run in parallel by the validate program.
+#
+# Author  :  Galen Hollins
+# Updated :  2020-02-02
+#
+
+usage()
+{
+    echo
+    echo "USAGE:"
+    echo
+    echo "  validate-parallel [-d directory] [-n num-groups]"
+    echo
+    echo "  If no directory is specified, the current directory is used."
+    echo
+    echo "  If num-groups is specified, then the inputs are split up into"
+    echo "  num-groups groups.  Otherwise the inputs are split up into the"
+    echo "  same amount of groups as there are processing cores on this"
+    echo "  system."
+}
+
+##### Main
+
+PRODUCT_DIR=.
+NUM_GROUPS=
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -d | --directory )      shift
+                                PRODUCT_DIR=$1
+                                ;;
+        -n | --num-groups )     shift
+                                NUM_GROUPS=$1
+                                ;;
+        -h | --help )           usage
+                                exit
+                                ;;
+        * )                     usage
+                                exit 1
+    esac
+    shift
+done
+
+#
+# Check to make sure validate is on the path
+#
+VALIDATE_ON_PATH=`which validate | wc -l`
+if [[ "${VALIDATE_ON_PATH}" -eq "1" ]]; then
+  echo "  validate binary is available (`which validate`) [OK]"
+else
+  echo
+  echo "  ERROR: No 'validate' found on your path."
+  echo "         Please ensure this is on your environment PATH, before running this script."
+  echo "         NOTE: this script requires that the name of the binary is 'validate'."
+  echo
+  exit 1
+fi
+
+#
+# Check to make sure GNU parallel is on the path
+#
+PARALLEL_ON_PATH=`which parallel | wc -l`
+if [[ "${PARALLEL_ON_PATH}" -eq "1" ]]; then
+  echo "  GNU parallel binary is available (`which parallel`) [OK]"
+else
+  echo
+  echo "  ERROR: No GNU parallel found on your path."
+  echo "         Please ensure this is on your environment PATH, before running this script."
+  echo "         See also:  https://www.gnu.org/software/parallel"
+  echo
+  exit 1
+fi
+
+NUM_CORES=`getconf _NPROCESSORS_ONLN`
+if [[ "${NUM_GROUPS}" -eq "" ]]; then
+  echo "  Splitting up into groups, based on number of cores this machine has..."
+  NUM_GROUPS=`echo $NUM_CORES`
```

Comment:
@galenatjpl after testing, does it make sense to use all cores? or was it more efficient to use ~1/2 of the cores available?

Reply:
I updated the code to be half-cores by default, and committed this change (along with doc updates).